app/presenters/article_presenter.rb

```diff
@@ -1,11 +1,42 @@
 # frozen_string_literal: true
 
 class ArticlePresenter
+  include ActionView::Helpers
+  include Rails.application.routes.url_helpers
+  include ActionView::Context
+
   def initialize(article)
     @article = article
   end
 
   def all_tags
     @article.tags.present? ? @article.tags.pluck(:name).join(', ') : ''
   end
+
+  def truncate_article_text
+    truncate(@article.text.to_s, escape: false, length: length_truncate) do
+      link_to 'Continue', controller: 'articles', action: 'show', id: @article.id
+    end
+  end
+
+  REGEX_FIGURE = /<figure[^>]/
```

Comment:
constants always locates after includes in the beginning of the file

Reply:
have you ran here rubocop?

Reply:
thanks. fixed